tests/__tests__/options.test.ts

```diff
@@ -717,6 +727,184 @@ describe('Middleware options', () => {
+      expect(
+        storeWithHandleSetAndPartializeAndEquality.temporal.getState()
+          .pastStates.length,
+      ).toBe(1);
+      vi.useRealTimers();
+    });
+
+    it('should not call throttle function if partialized state is unchanged according to diff fn', () => {
+      global.console.error = vi.fn();
+      vi.useFakeTimers();
+      const throttleIntervalInMs = 1000;
+      const storeWithHandleSetAndPartializeAndDiff = createVanillaStore({
+        handleSet: (handleSet) => {
+          return throttle<typeof handleSet>(
+            (state) => {
+              // used for determining how many times `handleSet` is called
+              console.error('handleSet called');
+              handleSet(state);
+            },
+            throttleIntervalInMs,
+            // Call throttle only on leading edge of timeout
+            { leading: true, trailing: false },
+          );
+        },
+        partialize: (state) => ({
+          count: state.count,
+        }),
+        diff: (pastState, currentState) => {
+          const myDiff = diff(currentState, pastState);
+          const newStateFromDiff = myDiff.reduce(
+            (acc, difference) => {
+              type State = typeof acc;
+              type Key = keyof State;
+              if (difference.type === 'CHANGE') {
+                const pathAsString = difference.path.join('.') as Key;
+                const value = difference.value;
+                acc[pathAsString] = value;
+              }
+              return acc;
+            },
+            {} as Partial<typeof currentState>,
+          );
+          return isEmpty(newStateFromDiff) ? null : newStateFromDiff;
+        },
+      });
+
+      const { incrementCountOnly, incrementCount2Only } =
+        storeWithHandleSetAndPartializeAndDiff.getState();
+      // Increment value not included in partialized state
+      act(() => {
+        incrementCount2Only();
+      });
+      // Proxy for determining how many times `handleSet` is called.
+      // handleSet should not be called if partialized state is unchanged
+      expect(console.error).toHaveBeenCalledTimes(0);
+      expect(
+        storeWithHandleSetAndPartializeAndDiff.temporal.getState().pastStates
+          .length,
+      ).toBe(0);
+      // Advance timer to be within throttle interval
+      vi.advanceTimersByTime(throttleIntervalInMs / 2);
+      act(() => {
+        incrementCountOnly();
+      });
+      // Count is in partialized state, so handleSet should have been called
+      expect(console.error).toHaveBeenCalledTimes(1);
+      // The first instance of a partialized state changing should add to history
+      expect(
+        storeWithHandleSetAndPartializeAndDiff.temporal.getState().pastStates
+          .length,
+      ).toBe(1);
+      vi.useRealTimers();
+    });
+
+    it('should always call throttle function on any partialized or non-partialized state change if no equality or diff fn is provided', () => {
```

Comment:
This test passes on `main` branch, but seemed helpful to explicitly define and to call out.

See this gif for what this looks like (notice useless history):

Sandbox: https://codesandbox.io/p/sandbox/zundo-handleset-partialize-with-no-equality-fn-and-no-diff-fn-zvs3sx?file=%2Fsrc%2FApp.tsx%3A23%2C37

Gif: 
![zundo-handleSet-partialize-no-equality-no-diff](https://github.com/charkour/zundo/assets/6432833/8d96310c-2876-475f-8394-291fa3eaaec7)